### DISCUSSION TITLE
How to contribute to this project?

### DISCUSSION THREAD

**Discussion by abhi1992** [JetBrains/lets-plot#627] [Q&A]
hey,
I would love to contribute to this project. It is something I love like kotlin + graphs. It's absolutely useless to build my own chart library as you have made tremendous progress here.
So, I want to help in contributing to this project, I am a software developer for 7+ years and I am comfortable in any new languages as long as it's not purely functional ( I hate them. ). 

If you could suggest me some bugs that I can start off on I can take off from there and see where I can land.

Thanks in advance,
Abhishek Banerjee

**Comment by alshan** [comment]
Hi Abhishek!

How about this feature request: https://github.com/JetBrains/lets-plot/issues/596 ?

For reference see also the lets-plot Kotlin variant of [`ggsave()`](https://lets-plot.org/kotlin/-lets--plot--kotlin/org.jetbrains.letsPlot.export/ggsave.html) which already supports PNG and JPEG export.

For the work you will need to clone this repository and setup a local dev environment. Don't be shy to ask any questions. 

Another candidate to take a look at is this future request: https://github.com/JetBrains/lets-plot/issues/76 which has been pushed down the road for quite a long time.

---

Also, we very much appreciate any sort of interesting content which is created with lets-plot and published.

---

In [lets-plot-kotlin](https://github.com/JetBrains/lets-plot-kotlin) some of the API still has insufficient documentation or no documentation.

---
As you can see you can help in various ways )

**Reply by abhi1992** [reply]
Hey, thanks a lot for the reply. And I am really honoured to be a part of this. I was looking around the code base for both the features. And could make some progress and understand a bit. I maybe wrong with my interpretation though since all I did was chase the rabbit hole and dumping conclusions. But here's what I found for issue #76 :

interface PointShape {
    val code: Int

    fun size(dataPoint: DataPointAesthetics): Double

    fun strokeWidth(dataPoint: DataPointAesthetics): Double

//    fun create(location: DoubleVector, dataPoint: DataPointAesthetics): SvgSlimObject
}

We have set PointShape.code as integer. Is it the cause we are unable to see the strings? 
Also, is there a standard email or help channel I can use rather than a public github disussion for this? 
Thanks in Advance!

**Reply by alshan** [reply]
This issue is a bit tricky. You have to look at the `ShapeOptionConverter`. 
Out of ggplot2 specs for the [`shape` aesthetic](https://ggplot2.tidyverse.org/articles/ggplot2-specs.html?q=shape#sec:shape-spec) we only support 
> - An integer in [0,25]

> - A . to draw the smallest rectangle that is visible, usualy 1 pixel.

The idea in https://github.com/JetBrains/lets-plot/issues/76 is to add support for few mnemonics, like "+" for shape 3.
Note, that such mnemonics are not in ggplot2 specs, but they can be found in Matplotlib  (see link in the issue).
